feature/home/presentation/src/main/java/com/sanaa/presentation/screen/mediaTabScreen/continueWatchingScreen/ContinueWatchingMediaScreenViewModel.kt

```diff
@@ -0,0 +1,201 @@
+package com.sanaa.presentation.screen.mediaTabScreen.continueWatchingScreen
+
+import androidx.paging.PagingData
+import androidx.paging.PagingSource
+import com.sanaa.presentation.BaseViewModel
+import com.sanaa.presentation.base.BasePagingSourceForHome
+import com.sanaa.presentation.screen.mediaTabScreen.MediaTabScreenEffect
+import com.sanaa.presentation.screen.mediaTabScreen.MediaTabScreenInteractionListener
+import com.sanaa.presentation.screen.mediaTabScreen.MediaTabScreenUiState
+import com.sanaa.presentation.state.MediaItem
+import com.sanaa.presentation.state.MediaType
+import com.sanaa.presentation.state.mapper.toState
+import entity.Movie
+import entity.TvSeries
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import usecase.ManageMovieUseCase
+import usecase.ManageTvSeriesUseCase
+import usecase.history.ManageWatchedMediaHistoryUseCase
+import usecase.history.ManageWatchedMediaHistoryUseCase.MediaHistoryItem
+
+class ContinueWatchingMediaScreenViewModel(
+    private val manageMovieUseCase: ManageMovieUseCase,
+    private val manageTvSeriesUseCase: ManageTvSeriesUseCase,
+    private val manageWatchedMediaHistoryUseCase: ManageWatchedMediaHistoryUseCase,
+    private val dispatcher: CoroutineDispatcher = Dispatchers.IO
+) : BaseViewModel<MediaTabScreenUiState, MediaTabScreenEffect>(
+    MediaTabScreenUiState(),
+    dispatcher
+), MediaTabScreenInteractionListener {
+
+    init {
+        fetchMovieGenres()
+        fetchTvShowGenres()
+        fetchMovies()
+        fetchTvShows()
+    }
+
+    private fun fetchMovies(genreId: Int? = null) {
+        tryToExecute(
+            callee = {
+                loadTopRatedMovies(genreId = genreId)
+            }, onSuccess = { mediaList ->
+                updateState {
+                    it.copy(movieList = mediaList, isLoading = false)
+                }
+            },
+            onError = { exception ->
```

Comment:
This is repetitive throughout the class:
`updateState {
                    it.copy(error = exception.message, isLoading = false)
}`

Maybe you need to move it inside a function and use it:
```suggestion
            onError =::onErrorOccurre()
```